operations/fetch.go

```diff
@@ -263,16 +264,28 @@ func cloneSource(task *service.RestTask, project *model.ProjectRef, config *mode
 	if variant == nil {
 		return errors.Errorf("couldn't find build variant '%v' in config", task.BuildVariant)
 	}
+	mfest, err := manifest.FindFromVersion(task.Version, task.Project, task.Revision)
+	if err != nil {
+		return errors.Errorf("error finding manifest for version '%s'", task.Version)
+	}
+	if mfest == nil && len(variant.Modules) != 0 { // manifest only required if variant has modules
+		return errors.Errorf("couldn't find manifest for version '%s'", task.Version)
+	}
+
 	for _, moduleName := range variant.Modules {
 		module, err := config.GetModuleByName(moduleName)
 		if err != nil || module == nil {
 			return errors.Errorf("variant refers to a module '%v' that doesn't exist.", moduleName)
 		}
+		mfestModule, ok := mfest.Modules[moduleName]
+		if !ok {
+			return errors.Errorf("couldn't find module '%s' in manifest", moduleName)
+		}
 		moduleBase := filepath.Join(cloneDir, module.Prefix, module.Name)
 		fmt.Printf("Fetching module %v at %v\n", moduleName, module.Branch)
 		err = clone(cloneOptions{
 			repo:     module.Repo,
-			revision: module.Branch,
+			revision: mfestModule.Revision,
```

Comment:
Don't we need to fall back to the HEAD revision in case it's not specified?

Reply:
I assumed this was always specified since we assume the module itself has to exist, but yes we can fall back.